### PR TITLE
Prepare Manual Testing

### PR DESCRIPTION
**Pull request by emaG3m**

### CODE REVIEW

File: golang/x/relay/client/cli/tx.go [hunk 1]
```diff
@@ -30,28 +32,49 @@ func GetTxCmd(storeKey string, cdc *codec.Codec) *cobra.Command {
...
 			cliCtx := context.NewCLIContext().WithCodec(cdc)
-
 			txBldr := auth.NewTxBuilderFromCLI().WithTxEncoder(utils.GetTxEncoder(cdc))
 
-			// note: unmarshalling json here is undesirable, but necessary
+			type IngestHeadersReq struct {
+				Headers []types.BitcoinHeader `json:"headers"`
+			}
+
+			var requestParams IngestHeadersReq
```
Comment (emaG3m) [inline golang/x/relay/client/cli/tx.go]: :(

File: golang/x/relay/client/cli/tx.go [hunk 2]
```diff
@@ -30,28 +32,49 @@ func GetTxCmd(storeKey string, cdc *codec.Codec) *cobra.Command {
...
-			jsonErr := json.Unmarshal([]byte(args[0]), &headers)
-			if jsonErr != nil {
-				return jsonErr
+			if viper.GetBool("inputfile") {
+				jsonFile, err := ioutil.ReadFile("scripts/seed_data/" + args[0])
+				if err != nil {
+					return err
+				}
+				jsonErr := json.Unmarshal([]byte(jsonFile), &requestParams)
+				headers = requestParams.Headers
```
Comment (emaG3m) [inline golang/x/relay/client/cli/tx.go]: :( feels redundant

File: golang/scripts/json_data/3_filled_requests.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[
+    {
+        "inputIndex": 0,
+        "outputIndex": 0,
+        "id": "01"
```
Comment (emaG3m) [inline golang/scripts/json_data/3_filled_requests.json]: do I represent `[]byte{1}` in json????

File: golang/x/relay/keeper/requests.go [hunk 4]
```diff
@@ -121,7 +133,7 @@ func (k Keeper) getNextID(ctx sdk.Context) (types.RequestID, sdk.Error) {
 	store := k.getRequestStore(ctx)
 	idTag := []byte(types.RequestIDTag)
 	if !store.Has(idTag) {
-		store.Set(idTag, bytes.Repeat([]byte{0}, 8))
+		store.Set(idTag, bytes.Repeat([]byte{1}, 8))
```
Comment (emaG3m) [inline golang/x/relay/keeper/requests.go]: stared with 1, do avoid doing an extra 0 bytes check.

File: golang/scripts/json_data/3_filled_requests.json (hunk 3 above)
Comment (emaG3m) [reply golang/scripts/json_data/3_filled_requests.json]: errors include
"01"
```
ERROR: Expected 8 bytes, got 1 bytes
```
"0000001"
```
"ERROR: encoding/hex: odd length hex string"
```

File: golang/x/relay/keeper/requests.go (hunk 4 above)
Comment (prestwich) [reply golang/x/relay/keeper/requests.go]: ```bytes.Repeat([]byte{1}, 8))```

will end up with `1` repeated 8 times. `[1, 1, 1, 1, 1, 1, 1, 1]` which corresponds to hex `0x0101010101010101`

If you want to start with 1 do 
```
next := make([]byte, 8)
next[7] = 1
```

I don't think you need to change this tho. Where is the extra 0 bytes check?

File: golang/scripts/json_data/3_filled_requests.json (hunk 3 above)
Comment (prestwich) [reply golang/scripts/json_data/3_filled_requests.json]: There is an implementation of marshall and unmarshall over here:

https://github.com/summa-tx/relays/blob/golang/golang/x/relay/types/requests.go#L66-L87

These should be represented as hex strings encoding 8 bytes.  So 1 is encoded as: `"0x0000000000000001"`

File: golang/x/relay/keeper/requests.go (hunk 4 above)
Comment (emaG3m) [reply golang/x/relay/keeper/requests.go]: Maybe this has to do with changing the datatype but if I have a 0 id, this is the error I get when handing in `0x`:
```
➜  golang git:(issue28) ✗ relaycli tx relay provideproof  1_check_proof.json 3_filled_requests.json  --from me  --inputfile --broadcast-mode block
ERROR: Expected 8 bytes, got 0 bytes
```

Comment (prestwich) [reply golang/x/relay/keeper/requests.go]: hand in `0x0000000000000000`. It expects these to be 8-bytes long regardless of whether those bytes are 0

File: golang/x/relay/types/msgs.go [hunk 5]
```diff
@@ -179,10 +179,10 @@ func (msg MsgNewRequest) Type() string { return "new_request" }
 // ValidateBasic runs stateless validation
 func (msg MsgNewRequest) ValidateBasic() sdk.Error {
 	// TODO: validate output types
-	if len(msg.Spends) != 36 {
+	if len(msg.Spends) != 36  && len(msg.Spends) != 0 {
 		return ErrSpendsLength(DefaultCodespace)
 	}
-	if len(msg.Pays) > 50 {
+	if len(msg.Pays) > 50 && len(msg.Spends) != 0 {
```
Comment (prestwich) [inline golang/x/relay/types/msgs.go]: should read `if len(msg.Pays) > 50`.  No need for the second check at all

Comment (emaG3m) [reply golang/x/relay/types/msgs.go]: 😂

**Comment by emaG3m** [comment]
Addressed https://github.com/summa-tx/relays/issues/52 in this PR. Flags from `gentx` `gentx-collect` had more to do with where to output chain data, than where to collect it. So just incorporated  a `scripts/json_data/genesis.json` file where other CLI json lives so it's accessible to users of this app

**Pull request merged**